### DISCUSSION TITLE
Sliding window inference overlap in 2D patches with 3 spatial dims (one of them being 1)

### DISCUSSION THREAD

**Discussion by YerePhy** [Project-MONAI/MONAI#4210] [Q&A] [answered]
**Question**
- How does the `monai.inferers.sliding_window_inference` `overlap` parameter interact with patches 2D? 

If the `predictor` eats patches of size (1, C, H, W, 1) and pops patches of the same size, there is no possible overlap across axis -1 or am I missing something?

**Comment by Nic-Ma** [comment, marked as answer]
Hi @YerePhy ,

Thanks for your interest here.
The `overlap` arg means: Amount of overlap between the sliding **windows**:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/inferer.py#L100

Thanks.

**Reply by YerePhy** [reply]
Hi, thanks in advance for your answer

Yes, I've read that, but how you define overlap (please note that I'm talking about overlap here as a concept, not about `overlap` parameter) between sliding windows if their size is for example (H, W, 1)? I mean, the windows can overlap across  axis 0 or 1 but not across axis 2, does `monai.inferers.sliding_window_inference` take that into account? 

Look this example, supose the shape of the window is (H, W, 1):

![Rectangle-#2](https://user-images.githubusercontent.com/69209435/166153072-74d6edbd-cac9-41b3-a188-2f954e5baa23.png)

**Reply by Nic-Ma** [reply]
Hi @YerePhy ,

For more details about the overlap & scan window interval, you can refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/utils.py#L166

Thanks.

**Reply by YerePhy** [reply]
Thanks @Nic-Ma, 

The answer is yes, because `interval` is 1 if `roi_size` is 1. 

It would be great to contribute to MONAI one of these days, but maybe I do not have enough experience yet.

**Reply by Nic-Ma** [reply]
Welcome your contribution!

Thanks.
